app/controllers/fhir/r4/api_controller.rb

```diff
@@ -210,6 +210,15 @@ def create
     when 'patient'
       return if doorkeeper_authorize!(*PATIENT_WRITE_SCOPES)
 
+      # Duplicate detection check
+      if request.headers['If-None-Exist'].present?
+        matches = search_patients(Rack::Utils.parse_nested_query(request.headers['If-None-Exist']))
+        num_matches = matches.size
+        status_ok(matches.first.as_fhir) && return if num_matches == 1
```

Comment:
I think @Bialogs is correct in pointing out the spec doesn't actually say to return the resource as FHIR. So we should just return a `200 OK` here.

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1098/commits/dfabb5dca271e1d1f47b09680a1dd63cd63f6218#diff-e79e38176ee697a8541cffba23c60d9decaa69382ca5c6e875da033508e5b428R217